keylime/common/algorithms.py

```diff
@@ -12,47 +12,30 @@ def is_accepted(algorithm: str, accepted: List[Any]) -> bool:
     return algorithm in accepted
 
 
-def _hashit(algorithm: str, data: bytes) -> bytes:
-    if algorithm == "sha1":
-        return hashlib.sha1(data).digest()
-    if algorithm == "sha256":
-        return hashlib.sha256(data).digest()
-    if algorithm == "sha384":
-        return hashlib.sha384(data).digest()
-    if algorithm == "sha512":
-        return hashlib.sha512(data).digest()
-    if algorithm == "sm3_256":
-        # SM3 is not guaranteed to be there
-        return hashlib.new("sm3", data).digest()
-
-    raise ValueError(f"Unsupported hash algorithm {algorithm}")
-
-
 class Hash(str, enum.Enum):
     SHA1 = "sha1"
     SHA256 = "sha256"
     SHA384 = "sha384"
     SHA512 = "sha512"
-    SM3_256 = "sm3_256"
+    SM3_256 = "sm3"
 
-    def __init__(self, *args):  # pylint: disable=unused-argument
-        super().__init__()
-        # Test hash to raise ValueError for unsupported hashes
-        _hashit(self.value, b"")
+    @classmethod
+    def from_algorithm(cls, algorithm: str) -> "Hash":
+        return cls(algorithm)
```

Comment:
So you don't want to also break right in the constructor when someone grabs the Hash("sm3") and the underlying crypto lib doesn't support it? Hash("bad") throws a ValueError and Hash("sm3") should throw one too IMO if the crypto lib doesn't support it because it looks like the usage of this class is to call `is_recognized()` first to see whether it will work.